src/shared/site-download.http.ts

```diff
@@ -0,0 +1,428 @@
+  setCookie(...cookies: Array<{ name: string; value: string; domain: string; path?: string }>): Promise<void>;
+  setUserAgent(userAgent: string): Promise<void>;
+  setViewport(viewport: { width: number; height: number; deviceScaleFactor?: number }): Promise<void>;
+};
+
+// Type definitions
+export type BodySchema = {
+  url: string;
+  cookies?: Array<{
+    name: string;
+    value: string;
+    domain?: string;
+    path?: string;
+    expires?: number;
+    httpOnly?: boolean;
+    secure?: boolean;
+    sameSite?: 'Strict' | 'Lax' | 'None';
+  }>;
+  headers?: Record<string, string>;
+  gotoOptions?: {
+    timeout?: number;
+    waitUntil?: 'load' | 'domcontentloaded' | 'networkidle0' | 'networkidle2';
+  };
+  waitForSelector?: {
+    selector: string;
+    timeout?: number;
+    visible?: boolean;
+    hidden?: boolean;
+  };
+  userAgent?: string;
+  viewport?: {
+    width: number;
+    height: number;
+    deviceScaleFactor?: number;
+  };
+  useExistingSession?: boolean;
+  sessionId?: string;
+};
+
+export type QuerySchema = SystemQueryParameters & {
+  token: string;
+};
+
+export type ResponseSchema = {
+  success: boolean;
+  error?: string;
+};
+
+// Schema validation objects
+export const BodySchema = {
+  type: 'object',
+  properties: {
+    url: {
+      type: 'string',
+      description: 'URL of the document to download',
+    },
+    cookies: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          name: { type: 'string' },
+          value: { type: 'string' },
+          domain: { type: 'string' },
+          path: { type: 'string' },
+          expires: { type: 'number' },
+          httpOnly: { type: 'boolean' },
+          secure: { type: 'boolean' },
+          sameSite: { 
+            type: 'string',
+            enum: ['Strict', 'Lax', 'None']
+          },
+        },
+        required: ['name', 'value'],
+      },
+    },
+    headers: {
+      type: 'object',
+      additionalProperties: { type: 'string' }
+    },
+    gotoOptions: {
+      type: 'object',
+      properties: {
+        timeout: { type: 'number' },
+        waitUntil: {
+          type: 'string',
+          enum: ['load', 'domcontentloaded', 'networkidle0', 'networkidle2']
+        }
+      }
+    },
+    waitForSelector: {
+      type: 'object',
+      properties: {
+        selector: { type: 'string' },
+        timeout: { type: 'number' },
+        visible: { type: 'boolean' },
+        hidden: { type: 'boolean' }
+      },
+      required: ['selector']
+    },
+    userAgent: { type: 'string' },
+    viewport: {
+      type: 'object',
+      properties: {
+        width: { type: 'number' },
+        height: { type: 'number' },
+        deviceScaleFactor: { type: 'number' }
+      },
+      required: ['width', 'height']
+    },
+    useExistingSession: { type: 'boolean' },
+    sessionId: { type: 'string' }
+  },
+  required: ['url']
+} as const;
+
```

Comment:
There is a build step that "explodes" these schemas into OpenAPI compatible ones. You shouldn't have to do this:

```js
export const QuerySchema = {
  type: 'object',
  properties: {
    token: { type: 'string' }
  },
  required: ['token']
} as const;
```

Instead, it can be just this:

```js
export type QuerySchema = SystemQueryParameters & {
  token: string;
};
```

Unless there was a reason to do so?

Reply:
Yeah it's a bit of fun code in a build script, helps with the amount of verbose changes for open-api